### DISCUSSION TITLE
Debugging xtask

### DISCUSSION THREAD

**Discussion by FawazTirmizi** [oxidecomputer/hubris#602] [Q&A] [answered]
I've been making some changes to different parts of the repository, including xtask, and am running into some bugs with the linker script generation. Is there a way for xtask to be debugged in GDB? When I try by running `rust-gdb --args target/debug/xtask dist app/test-app/app.toml`, I get the error `Error: environment variable not found`.

**Comment by mx-shift** [comment, marked as answer]
xtask expects to be run via `cargo run` which sets a few environment variables.  One possible way to run xtask under gdb would be to follow the instructions at https://stackoverflow.com/questions/52685869/is-there-a-way-to-directly-run-the-program-built-by-cargo-in-gdb-or-lldb.

**Reply by FawazTirmizi** [reply]
That works, thanks! Here is what I ended up doing:

`.config/cargo.toml`:
```
[target.x86_64-unknown-linux-gnu]
runner = ["/home/fawaz/cargo_runner.sh"]
```

`cargo_runner.sh`:
```
#!/bin/bash

if [[ -z $CARGO_DEBUG ]]; then
    exec $*
else
    exec rust-gdb --args $*
fi
```
